src/components/CreditCardForm/CreditCardForm.tsx

```diff
@@ -0,0 +1,45 @@
+import './style.css'
+import { useRef, ChangeEvent, useEffect } from 'react'
+
+export const CreditCardForm = (): JSX.Element => {
+  const inputRef1 = useRef<HTMLInputElement>(null!)
+  const inputRef2 = useRef<HTMLInputElement>(null!)
+  const inputRef3 = useRef<HTMLInputElement>(null!)
+  const inputRef4 = useRef<HTMLInputElement>(null!)
+
+  const inputRefs = [inputRef1, inputRef2, inputRef3, inputRef4]
+
+  useEffect(() => {
+    inputRefs[0].current.focus()
+  }, [])
+
+  const handleInput = (
+    e: ChangeEvent<HTMLInputElement>,
+    index: number
+  ): void => {
+    const newValue = e.target.value.replace(/\D/g, '')
```

Comment:
Tohle `/\D/g` bych si vytáhla do vlastní proměnné. Regular expressions můžou být i mnohem složitější než jenom vybrat čísla a stejně to všichni googlí. Kdyby to náhodou nefungovalo, tak to zkus něco jako `const numbersRegExp = new RegExp("/\D/g");` (viz [tady](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp)).